com.trekglobal.idempiere.rest.api/src/com/trekglobal/idempiere/rest/api/json/ExpandParser.java

```diff
@@ -165,15 +214,18 @@ private ModelHelper getModelHelper(List<String> operators, String tableName, Str
 	}
 	
 	private String getFilterClause(List<String> operators, String keyColumnName, int keyColumnValue) {
-		String filterClause = keyColumnName + " eq " + keyColumnValue; 
+		StringBuilder filterClause = new StringBuilder(keyColumnName + " eq " + keyColumnValue);
+		
+		if (isRecordIDTableIDFK(keyColumnName))
+			filterClause.append(" AND " + TABLE_ID_COLUMN + " eq " + po.get_Table_ID());
 		
 		for (String operator : operators) {
 			if (operator.startsWith(QueryOperators.FILTER)) {
-				filterClause = filterClause + " AND " + getStringOperatorValue(operator);
+				filterClause.append(filterClause + " AND " + getStringOperatorValue(operator));
```

Comment:
there is a problem here that broke the test case for ```c_order?$select=DocumentNo,Description&$expand=C_OrderTax,C_OrderLine($select=Line,Linenetamt ; $top=1 ; $skip=1 ; $filter=IsActive eq true ; $orderby=Line)```

It must be fixed with this patch:
[pr136PR.txt](https://github.com/bxservice/idempiere-rest/files/9972859/pr136PR.txt)

Can you please check?